File: api-frontend/apps/web-antd/src/api/system/notification.ts
import { requestClient } from '#/api/request';

export interface NotificationRecord {
  id: number;
  tenant_id: number;
  sender_id: number;
  sender_name: string;
  receiver_id: number;
  title: string;
  content: string;
  type: string;
  is_read: number;
  read_at: string;
  extra: string;
  reply_to_id: number;
  created_at: string;
}

export interface PageResult<T> {
  list: T[];
  total: number;
  page: number;
  page_size: number;
}

export function getNotificationList(params: {
  page: number;
  page_size: number;
  type?: string;
  is_read?: number;
}) {
  return requestClient.get<PageResult<NotificationRecord>>('/notifications', { params });
}

export function getUnreadCount() {
  return requestClient.get<{ count: number }>('/notifications/unread-count');
}

export function markAsRead(id: number) {
  return requestClient.put(`/notifications/${id}/read`);
}

export function markAllAsRead() {
  return requestClient.put('/notifications/read-all');
}

export function deleteNotification(id: number) {
  return requestClient.delete(`/notifications/${id}`);
}

export function sendNotification(data: {
  receiver_ids?: number[];
  role_ids?: number[];
  title: string;
  content?: string;
  type?: string;
}) {
  return requestClient.post('/notifications/send', data);
}

export function getNotificationDetail(id: number) {
  return requestClient.get<{
    notification: NotificationRecord;
    replies: NotificationRecord[];
  }>(`/notifications/${id}`);
}

export function replyNotification(id: number, content: string) {
  return requestClient.post(`/notifications/${id}/reply`, { content });
}

export function deleteReply(id: number) {
  return requestClient.delete(`/notifications/reply/${id}`);
}

File: api-backend/internal/model/notification.go
package model

import "time"

// Notification 站内消息表
type Notification struct {
	TenantBaseModel
	SenderID   uint       `gorm:"default:0" json:"sender_id"`   // 0=系统消息
	ReceiverID uint       `gorm:"index;not null" json:"receiver_id"`
	Title      string     `gorm:"size:200;not null" json:"title"`
	Content    string     `gorm:"type:text" json:"content"`
	Type       string     `gorm:"size:50;default:system" json:"type"` // system/task/message
	IsRead     int8       `gorm:"default:0" json:"is_read"`          // 0=未读 1=已读
	ReadAt     *time.Time `json:"read_at"`
	Extra      string     `gorm:"type:text" json:"extra"`      // JSON扩展字段
	ReplyToID  uint       `gorm:"default:0" json:"reply_to_id"` // 回复哪条消息，0=原始消息
	SenderName string     `gorm:"-" json:"sender_name"`         // 发送者名称（非DB字段）
}

func (Notification) TableName() string {
	return "notifications"
}
